### PR TITLE
drivers/flash/qspi_nor_flash: Remove dev null check

### PR DESCRIPTION
**Pull request by de-nordic**
The PR contains two commits:

 - e33123b2f1 drivers/flash/nrf_qspi_nor: Remove qspi_nor_read_id param flash_id;
 - ef6d36e1ec drivers/flash/qspi_nor_flash: Remove dev null check.
 
Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

File: drivers/flash/nrf_qspi_nor.c [hunk 1]
```diff
@@ -1109,7 +1106,9 @@ static int qspi_nor_write_protection_set(const struct device *dev,
  */
 static int qspi_nor_configure(const struct device *dev)
 {
-	const struct qspi_nor_config *params = dev->config;
+	if (!dev) {
+		return -ENXIO;
+	}
```
Comment (anangl) [inline drivers/flash/nrf_qspi_nor.c:1112]: Unnecessary empty line.

File: drivers/flash/nrf_qspi_nor.c [hunk 2]
```diff
@@ -1109,7 +1106,9 @@ static int qspi_nor_write_protection_set(const struct device *dev,
...
```
Comment (anangl) [inline drivers/flash/nrf_qspi_nor.c:1111]: Other drivers don't perform such checks, so perhaps this should be removed completely?

File: drivers/flash/nrf_qspi_nor.c [hunk 3]
```diff
@@ -1119,6 +1118,8 @@ static int qspi_nor_configure(const struct device *dev)
 
 	ANOMALY_122_UNINIT(dev);
 
+	const struct qspi_nor_config *params = dev->config;
```
Comment (anangl) [inline drivers/flash/nrf_qspi_nor.c:1121]: What about moving it to `qspi_nor_read_id()` where it is actually needed?

File: drivers/flash/nrf_qspi_nor.c (hunk 2 above)
Comment (de-nordic) [reply drivers/flash/nrf_qspi_nor.c:1111]: I was also considering that or CHECKIF. I will remove it.

File: drivers/flash/nrf_qspi_nor.c (hunk 3 above)
Comment (de-nordic) [reply drivers/flash/nrf_qspi_nor.c]: Done

File: drivers/flash/nrf_qspi_nor.c (hunk 2 above)
Comment (de-nordic) [reply drivers/flash/nrf_qspi_nor.c]: Done

File: drivers/flash/nrf_qspi_nor.c (hunk 1 above)
Comment (de-nordic) [reply drivers/flash/nrf_qspi_nor.c]: Done

**Review by anangl — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**